Sources/Sentry/SentryClient.m

```diff
@@ -314,18 +315,18 @@ - (SentryId *)sendEvent:(SentryEvent *)event
 - (nullable SentryTraceContext *)getTraceStateWithEvent:(SentryEvent *)event
                                               withScope:(SentryScope *)scope
 {
-    id<SentrySpan> span;
+    SentryTracer *tracer;
     if ([event isKindOfClass:[SentryTransaction class]]) {
-        span = [(SentryTransaction *)event trace];
+        tracer = ((SentryTransaction *)event).trace;
     } else {
         // Even envelopes without transactions can contain the trace state, allowing Sentry to
         // eventually sample attachments belonging to a transaction.
-        span = scope.span;
+        tracer = scope.span.tracer;
     }
 
-    SentryTracer *tracer = [SentryTracer getTracer:span];
-    if (tracer == nil)
+    if (tracer == nil) {
         return nil;
+    }
```

Comment:
This entire thing would be much cleaner this way.

Reply:
This make sense